### PR TITLE
latex layer: add company-reftex

### PR DESCRIPTION
**Pull request by et2010**
https://github.com/TheBB/company-reftex

This is very handy. Thanks! @TheBB

### CODE REVIEW

File: layers/+lang/latex/packages.el [hunk 1]
```diff
@@ -144,13 +145,20 @@
     :defer t
```
Comment (Compro-Prasad) [inline layers/+lang/latex/packages.el]: Remove this whole function as you are redefining it later.

File: layers/+lang/latex/packages.el [hunk 2]
```diff
@@ -144,13 +145,20 @@
...
     :init (spacemacs|add-company-backends
             :backends
-            company-auctex-labels
```
Comment (Compro-Prasad) [inline layers/+lang/latex/packages.el]: Why remove these lines?

Comment (et2010) [reply layers/+lang/latex/packages.el]: They are replaced by their company-reftex counterparts:

`company-reftex-labels` and `company-reftex-citations`

**Comment by siddharthasahu** [comment]
Hello. I just tried this patch and I can now get autocompletion inside my citations and references. I'm not sure why the tests are failing but pinging to consider merging this to master. Thanks!

**Comment by duianto** [comment]
Thank you for contributing to Spacemacs!
The PR has been cherry-picked into the `develop` branch, you can safely delete your branch.
Commit: https://github.com/syl20bnr/spacemacs/commit/2e6e69eeff2d00e594c58985655563d27568bc4f

@siddharthasahu Thanks for testing this.

**Pull request closed without merging**